heron/api/src/java/com/twitter/heron/api/tuple/Tuple.java

```diff
@@ -29,7 +30,7 @@
  * use another type, you'll need to implement and register a serializer for that type.
  * @see <a href="https://storm.apache.org/documentation/Serialization.html">Storm serialization</a>
  */
-public interface Tuple {
+public interface Tuple extends Serializable {
```

Comment:
Is this a breaking change? @maosongfu?

Reply:
it does not break current code. But I am wondering why we need this change?

Reply:
@maosongfu The change is need because we need a way to store buffered tuples in windows that haven't been triggered yet.  So that when a checkpoint a happens, the state of all untriggered windows, which is essentially just buffered tuples, will be persisted. This is to maintain the effectively once guarantee which windowing components.